sda-commons-web-autoconfigure/src/test/java/org/sdase/commons/spring/boot/web/metadata/ApiClientConfigurationTest.java

```diff
@@ -0,0 +1,98 @@
+ */
+package org.sdase.commons.spring.boot.web.metadata;
+
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.springframework.boot.test.context.SpringBootTest.WebEnvironment.RANDOM_PORT;
+import static org.springframework.http.MediaType.APPLICATION_JSON;
+
+import java.util.List;
+import java.util.Map;
+import org.junit.jupiter.api.Test;
+import org.junitpioneer.jupiter.SetSystemProperty;
+import org.sdase.commons.spring.boot.web.auth.AuthTestApp;
+import org.sdase.commons.spring.boot.web.testing.auth.EnableSdaAuthMockInitializer;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers;
+
+/** Test that http client configuration is correct. */
+@SpringBootTest(
+    classes = {AuthTestApp.class, MetadataTestApp.class},
+    webEnvironment = RANDOM_PORT,
+    // verify authentication in this test, not authorization
+    properties = {
+      "opa.disable=true",
+      "metadata.other.baseUrl=http://localhost:${wiremock.server.port}/api",
+      "metadata.otherAuthenticated.baseUrl=http://localhost:${wiremock.server.port}/api",
+    })
+@ContextConfiguration(initializers = EnableSdaAuthMockInitializer.class)
+@SetSystemProperty(key = "METADATA_FIELDS", value = "tenant-id")
+@AutoConfigureMockMvc
+class ApiClientConfigurationTest {
+  @Autowired private MockMvc mockMvc;
+
+  @MockBean private MetadataOtherServiceClient metadataOtherServiceClient;
+
+  @MockBean private MetadataOtherServiceAuthenticatedClient metadataOtherServiceAuthenticatedClient;
+
+  @Test
+  void submitMetadataContextInPlatformClient() throws Exception {
+    DetachedMetadataContext metadataContext = new DetachedMetadataContext();
+    metadataContext.put("tenant-id", List.of("t-1"));
+    MetadataContext.createContext(metadataContext);
+
+    var httpHeaders = new HttpHeaders();
+    httpHeaders.put("tenant-id", List.of("t-1"));
+
+    when(metadataOtherServiceAuthenticatedClient.getSomething(httpHeaders))
+        .thenReturn(Map.of("hello", "world"));
+
+    mockMvc
+        .perform(
+            MockMvcRequestBuilders.request(HttpMethod.GET, "/metadataAuthProxy")
+                .headers(httpHeaders)
+                .accept(APPLICATION_JSON))
+        .andExpect(MockMvcResultMatchers.status().isOk());
+    verify(metadataOtherServiceAuthenticatedClient).getSomething(httpHeaders);
+
+    MetadataContext.createContext(new DetachedMetadataContext());
```

Comment:
I'm not 100% familiar with our API. Don't we have a `clear` or `reset` method?
Should we move this line to a `beforeEach` method?

Reply:
We have it [for internal use](https://github.com/SDA-SE/sda-spring-boot-commons/blob/PLP-698-implement-for-rest-calls/sda-commons-web-autoconfigure/src/main/java/org/sdase/commons/spring/boot/web/metadata/MetadataContextHolder.java#L26). That would be accessible in this test. I would also prefer `@AfterEach` and/or `@BeforeEach`, but at least a `finally`.